### PR TITLE
fix(ui-react): merge internal and passed focus event handler props in Checkbox, remove extraneous useCheckbox hook

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
`Checkbox`:
- merge event handler props with internal handlers
- set `value` prop to optional
- remove `useCheckbox` hook and unit tests

Other:
- migrate  a `useTheme` unit test from _ThemeProvider.test.tsx_  to _useTheme.test.tsx_ to fix a Radix warning

#### Issue #, if available
NA

#### Description of how you validated changes
Manually tested, unit tests
#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/hooks/__tests__/useTheme.test.tsx [hunk 1]
```diff
@@ -30,6 +30,31 @@ describe('useTheme', () => {
     );
   });
 
+  it('extends the theme passed to `ThemeProvider`', () => {
```
Comment (calebpollman) [inline packages/react/src/hooks/__tests__/useTheme.test.tsx:33]: Moving here fixes this warning:
``` 
 Warning: Detected multiple renderers concurrently rendering the same context provider. This is currently unsupported.
          at $cc45c1b701a63adc$export$c760c09fdd558351 (/amplify-ui/packages/react/node_modules/@radix-ui/react-direction/dist/packages/react/direction/src/Direction.tsx:15:11)
          at AmplifyProvider (/amplify-ui/packages/react/src/components/ThemeProvider/ThemeProvider.tsx:44:3)
```

File: packages/react/src/primitives/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -26,26 +27,51 @@ const CheckboxPrimitive: Primitive<CheckboxProps, 'input'> = (
...
-  const initialChecked = checked !== undefined ? checked : defaultChecked;
 
-  const { dataChecked, dataFocus, onBlur, onChange, onFocus, setDataChecked } =
-    useCheckbox(initialChecked, onChangeProp);
+  const [focused, setFocused] = React.useState(false);
 
-  React.useEffect(() => {
-    const isControlled = checked !== undefined;
-    if (isControlled && checked !== dataChecked) {
-      setDataChecked(checked);
```
Comment (calebpollman) [inline packages/react/src/primitives/Checkbox/Checkbox.tsx:46]: Removed this effect, it is unnecessary to update the local checked state if the `Checkbox` is controlled

File: packages/react/src/primitives/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -109,12 +135,12 @@ const CheckboxPrimitive: Primitive<CheckboxProps, 'input'> = (
       ) : (
         <IconCheck
           className={iconClasses}
-          data-checked={dataChecked}
+          data-checked={checked}
```
Comment (dbanksdesign) [inline packages/react/src/primitives/Checkbox/Checkbox.tsx:138]: We can get rid of this

File: packages/react/src/primitives/Checkbox/Checkbox.tsx [hunk 4]
```diff
@@ -165,9 +191,9 @@ const CheckboxPrimitive: Primitive<CheckboxProps, 'input'> = (
         aria-hidden="true"
         as="span"
         className={flexClasses}
-        data-checked={dataChecked}
+        data-checked={checked}
```
Comment (dbanksdesign) [inline packages/react/src/primitives/Checkbox/Checkbox.tsx:194]: We can remove these data- attributes

**Review by dbanksdesign — APPROVED** [review]
LGTM! We can probably get rid of the data- attributes, but not necessarily in this PR

**Review by hbuchel — DISMISSED** [review]
Thanks for fixing this!

**Review by dbanksdesign — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]

**Pull request merged**